cmd/signals.go

```diff
@@ -18,32 +18,75 @@ package cmd
 
 import (
 	"os"
-	"os/signal"
 )
 
-// signalTrap traps the registered signals and notifies the caller.
-func signalTrap(sig ...os.Signal) <-chan bool {
-	// channel to notify the caller.
-	trapCh := make(chan bool, 1)
+func handleSignals() {
+	// Custom exit function
+	exit := func(err error) {
+		// If global profiler is set stop before we exit.
+		if globalProfiler != nil {
+			globalProfiler.Stop()
+		}
 
-	go func(chan<- bool) {
-		// channel to receive signals.
-		sigCh := make(chan os.Signal, 1)
-		defer close(sigCh)
+		rv := 0
+		if err != nil {
+			rv = 1
+		}
 
-		// `signal.Notify` registers the given channel to
-		// receive notifications of the specified signals.
-		signal.Notify(sigCh, sig...)
+		os.Exit(rv)
+	}
 
-		// Wait for the signal.
-		<-sigCh
+	stopProcess := func() error {
+		err := globalHTTPServer.Shutdown()
+		errorIf(err, "Unable to shutdown http server")
+		if objAPI := newObjectLayerFn(); objAPI != nil {
+			oerr := objAPI.Shutdown()
+			errorIf(oerr, "Unable to shutdown object layer")
+			if err == nil {
+				err = oerr
```

Comment:
This is pretty convoluted code.. why is it done this way? we should just return in this pattern 

```
if err := globalHTTPServer.Shutdown(); err != nil {
     errorIf(err, "Unable to shutdown http server")
     return err
}
```